### PR TITLE
NCL 6.4.0

### PR DESCRIPTION
**Pull request by khallock**
We are preparing to release NCL 6.4.0 and have created a 6.4.0 tag in the NCL repo. We will hopefully be announcing the release next week, and I would like to run this updated recipe through CircleCI/TravisCI in advance to ensure everything is working so that we can have a 6.4.0 conda-forge build available when we make our announcement.

Even if the checks for this PR succeed, please wait for an announcement on https://ncl.ucar.edu before merging. Thanks!

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,48 +24,50 @@ requirements:
...
         - jpeg 9*
         - zlib 1.2.*
         - hdf4
         - hdf5 1.8.17|1.8.17.*
         - libiconv
         - curl
-        - cairo  # [linux]
-        - freetype 2.6.*  # [linux]
+        - cairo 1.14.*  # [linux]
+        - freetype 2.7|2.7.*  # [linux]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Thanks for updating the pins!

**Comment by ocefpaf** [comment]
> Even if the checks for this PR succeed, please wait for an announcement on https://ncl.ucar.edu before merging. Thanks!

Sure. Feel free to self-merge updates BTW. That is fair game in conda-forge.

PS: Do we expect issues with the display on OS X? Will we need the x11 packages?

**Comment by khallock** [comment]
> Sure. Feel free to self-merge updates BTW. That is fair game in conda-forge.

Great! I wasn't sure if I needed someone else to verify the PR or if the CI checks are good enough.

> PS: Do we expect issues with the display on OS X? Will we need the x11 packages?

I believe the issues related to X11 on OS X began when conda-smithy updated the Travis CI build image from `osx_image: beta-xcode6.1` to `osx_image: xcode6.4`. I chose not to re-render the feedstock for this PR because nothing should have changed in terms of CI configuration and also because I know that it would cause problems with X11.

Obviously, we'll need to figure out a better solution in the future, but would it be alright to continue using `osx_image: beta-xcode6.1` for now in order to create a working release of NCL 6.4.0?

**Comment by khallock** [comment]
Well, it seems like we will still run into this X11 issue:
> This job was configured to run on an OS X image that was retired on January 20, 2017. It was routed to our Xcode 6.4 infrastructure.

I know that conda-forge feedstocks are configured to remove homebrew from the OS X build environment, but would it perhaps be possible to _not_ remove homebrew and instead use "brew cask" to install XQuartz?

**Comment by ocefpaf** [comment]
> I believe the issues related to X11 on OS X began when conda-smithy updated the Travis CI build image from `osx_image: beta-xcode6.1` to `osx_image: xcode6.4`. I chose not to re-render the feedstock for this PR because nothing should have changed in terms of CI configuration and also because I know that it would cause problems with X11.

The thing is that the old image is no longer available on Travis-CI... So with or without the re-rendering I am afraid we will face that issue :unamused: 

> I know that conda-forge feedstocks are configured to remove homebrew from the OS X build environment, but would it perhaps be possible to _not_ remove homebrew and instead use "brew cask" to install XQuartz?

The reason to remove homebrew is to avoid linking to packages that may not be available on the end users computers. However, pretty much like in the Linux case were we rely on `yum` for certain things, if we can prove that no linking will be broken we can try yo do exactly what you suggest. We need to discuss that though to ensure stability across the ecosystem. I will bring that up in our meetings... Maybe even write a conda-forge enhancement proposal for that.

**Comment by khallock** [comment]
Theoretically we could use "brew cask install xquartz" and then continue with removing homebrew. This "brew cask" command simply installs XQuartz from the .dmg hosted on the official XQuartz site:
https://github.com/caskroom/homebrew-cask/blob/master/Casks/xquartz.rb
https://www.xquartz.org/

The only new links that could potentially occur would be with the XQuartz-provided x11, cairo, etc. This is okay for NCL as it requires those packages anyway.

Alternatively, is there any possibility of conda-forge/cairo-feedstock#23 being resolved soon?

**Comment by khallock** [comment]
I have created another PR (conda-forge/ncl-feedstock#8) to test "brew cask install xquartz" to avoid cluttering this PR.

**Comment by khallock** [comment]
I just noticed that this successful check on OS X was using the xcode7.3 build image, so perhaps that's the reason why XQuartz was suddenly available again (without needing to use homebrew).

I just re-rendered the other PR (that uses homebrew to install XQuartz) to ensure we're using the appropriate build image.

**Comment by ocefpaf** [comment]
> I just noticed that this successful check on OS X was using the xcode7.3 build image, so perhaps that's the reason why XQuartz was suddenly available again (without needing to use homebrew).

Those is good news.

> I just re-rendered the other PR (that uses homebrew to install XQuartz) to ensure we're using the appropriate build image.

Any reason why we should not use this one instead? I am afraid that getting `XQuartz` from homebrew may cause broken linkage to end users.

**Comment by khallock** [comment]
Do you know if there would be any issues if we use the xcode7.3 image for this package when most other conda-forge packages are built using the xcode6.4 image?

My main concern would be the fact that the xcode6.4 image uses OS X 10.10, while xcode7.3 uses 10.11.

**Comment by ocefpaf** [comment]
> Do you know if there would be any issues if we use the xcode7.3 image for this package when most other conda-forge packages are built using the xcode6.4 image?

In principle I would be against that but,

- `ncl` is at the "end point" of most dependencies (I don't know anything that is built against it) so we don't expect linking issues ;
- if `ncl`+`xcode7.3` builds fine with the dependencies built against xcode6.4, then I am OK with that.

> My main concern would be the fact that the xcode6.4 image uses OS X 10.10, while xcode7.3 uses 10.11.

Yep. Same here. But feel free to test it and, if it works, you need to watch out for re-renderings to avoid `.travis.yml` getting "corrected" to `xcode6.4`.

**Comment by khallock** [comment]
I have discovered that when Travis CI announced their xcode7.3 image, they provided a list of software that is already installed in the build environment (as well as the installation method used): https://blog.travis-ci.com/2016-10-04-osx-73-default-image-live/

That link indicates that XQuartz was installed in the xcode7.3 image using `brew cask`.

The version of XQuartz installed by default in the xcode7.3 image is identical to what would be installed by running `brew cast install xquartz` in the xcode6.4 image, which are both identical to what a user would get if they downloaded and installed XQuartz directly from its [official webiste](https://www.xquartz.org/). Considering that existing builds of NCL and PyNGL _already require_ XQuartz to run (and this was seemingly okay since previous build images included it), would installing XQuartz manually using `brew cask` really be a serious concern?

Both solutions (`xcode6.4` + adding `brew cask install xquartz` vs `xcode7.3` + pre-installed xquartz via `brew cask`) would require "fixing" `.travis.yml` after re-rendering, but at least using xcode6.4 would ensure compatibility with other packages. (For the record, PyNGL is the only package I know of that depends on NCL.)

Once the updated cairo and the xorg-* packages are considered stable for OS X, we can move away from `brew cask`, but for now it seems like the best option.

**Comment by ocefpaf** [comment]
> The version of XQuartz installed by default in the xcode7.3 image is identical to what would be installed by running `brew cast install xquartz` in the xcode6.4 image, which are both identical to what a user would get if they downloaded and installed XQuartz directly from its [official webiste](https://www.xquartz.org/). Considering that existing builds of NCL and PyNGL _already require_ XQuartz to run (and this was seemingly okay since previous build images included it), would installing XQuartz manually using `brew cask` really be a serious concern?

I would start with that and create a long term plan to package `XQuartz` on conda-forge.

> Both solutions (`xcode6.4` + adding `brew cask install xquartz` vs `xcode7.3` + pre-installed xquartz via `brew cask`) would require "fixing" `.travis.yml` after re-rendering, but at least using xcode6.4 would ensure compatibility with other packages. (For the record, PyNGL is the only package I know of that depends on NCL.)

I would go for the `xcode6.4` solution for now.

> Once the updated cairo and the xorg-* packages are considered stable for OS X, we can move away from `brew cask`, but for now it seems like the best option.

Or our own `xquartz` not sure which one is preferred by OS X users.

**Comment by khallock** [comment]
> I would go for the xcode6.4 solution for now.

Sounds good, I'll merge the other PR. Thanks for all of your help with this, @ocefpaf!

**Pull request merged**